src/validator.ts

```diff
@@ -3,30 +3,41 @@ import {
   parseMessageWithCustomArgs,
 } from '@lemoncode/fonk';
 
-// TODO: Add validator type
-const VALIDATOR_TYPE = '';
+const VALIDATOR_TYPE = 'PREVIOUS_DATE';
 
-// TODO: Add default message
-let defaultMessage = '';
+export interface CustomArgs {
+  date: Date;
+}
+
+const MISSING_ARGS =
+  'FieldValidationError: date option for date validation is mandatory. Example: { date: new Date() }.';
+const BAD_PARAMETER = 'Value must be a valid Date object.';
+
+let defaultMessage = "Date isn't previous to the one provided.";
 export const setErrorMessage = message => (defaultMessage = message);
 
-const isDefined = value => value !== void 0 && value !== null && value !== '';
+export const validator: FieldValidationFunctionSync<CustomArgs> = ({
+  value,
+  message = defaultMessage,
+  customArgs,
+}) => {
+  if (!customArgs) {
+    throw new Error(MISSING_ARGS);
+  }
+
+  const { date } = customArgs;
 
-export const validator: FieldValidationFunctionSync = fieldValidatorArgs => {
-  const { value, message = defaultMessage, customArgs } = fieldValidatorArgs;
+  if (!(value instanceof Date)) {
+    throw new TypeError(BAD_PARAMETER);
+  }
 
-  // TODO: Add validator
-  const succeeded = !isDefined(value) || ...;
```

Comment:
We have to keep `!isDefined(value)` because if user wants to check it, he has to use `required` validator

Reply:
You want the validator to return a successful validation when passing values like undefined, null etc.?

Reply:
Added isDefined constant and !isDefined handling => cd08fccdc089f216eec3d2ac6576651f215cb698